### ISSUE TITLE
"Total motif counts" should just treat the graph as undirected

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1266]
[`motifs_randesu_no`](https://igraph.org/c/doc/igraph-Motifs.html#igraph_motifs_randesu_no) and [`motifs_randesu_estimate`](https://igraph.org/c/doc/igraph-Motifs.html#igraph_motifs_randesu_estimate) have the following note:

> Directed motifs will be counted in directed graphs and undirected motifs in undirected graphs.

Technically, this is true. This is what the internal implementation does.

However, counting "the total number of motifs" here refers to counting weakly connected subgraphs. It just counts the vertex triplets (size=3) or vertex quadruplets (size=4) which are weakly connected.

There is no difference between directed and undirected.

Furthermore, operating in undirected mode is faster. Here's a small benchmark in Mathematica (which includes an overhead, yet the undirected version is still faster):

```
In[46]:= IGMotifsTotalCount[g, 3] // RepeatedTiming
Out[46]= {0.0047, 99250}

In[47]:= IGMotifsTotalCount[g, 4] // RepeatedTiming
Out[47]= {0.035, 1307374}

In[48]:= ug = UndirectedGraph[g];

In[49]:= IGMotifsTotalCount[ug, 3] // RepeatedTiming
Out[49]= {0.0035, 99250}

In[50]:= IGMotifsTotalCount[ug, 4] // RepeatedTiming
Out[50]= {0.024, 1307374}
```

(Note: the output above is in the form `{timing, subgraphCount}`.)

Unless I am missing something, there would be a benefit in just switching over to undirected mode for all graphs and removing the note from the documentation.

----

BTW, can anyone point out what applications does counting connected `k`-tuples of vertices have?

**Comment by szhorvat** [comment]
@gaborcsardi I think you implemented this. Any objections to the above? Did I miss anything?

**Comment by gaborcsardi** [comment]
This seems to change the results, so we cannot change it.

**Comment by szhorvat** [comment]
My point was exactly that it doesn't change the result. Counting weakly connected triplet/quadruplets in a directed graph is the same as counting connected triplets/quadruplets in the undirected versions of the same graph.

Out[46] and Out[49] are the same in my example (but the latter was computed faster).

Out[47] and Out[50] are also the same.

**Comment by gaborcsardi** [comment]
Right, sure, go ahead then.

**Comment by szhorvat** [comment]
I looked at the code, and there appears to be nothing that takes edge directions into account. I believe that the speedup when converting to undirected was caused by reducing the number of edges. Indeed, if I first generate an undirected graph, then convert to directed with arbitrary edge orientations, the function will take the same amount of time to run on both.

I will simply clarify the documentation and close this issue.
